ui/app/components/namespace-picker.js

```diff
@@ -3,168 +3,79 @@
-  canAccessMultipleNamespaces: gt('accessibleNamespaces.length', 1),
-  isUserRootNamespace: computed('auth.authData.userRootNamespace', 'namespacePath', function () {
-    return this.auth.authData.userRootNamespace === this.namespacePath;
-  }),
-
-  namespaceDisplay: computed('namespacePath', 'accessibleNamespaces', 'accessibleNamespaces.[]', function () {
-    const namespace = this.namespacePath;
-    if (!namespace) {
-      return 'root';
-    }
-    const parts = namespace?.split('/');
-    return parts[parts.length - 1];
-  }),
+/**
+ * @module ApplicationComponent component is used to display a dropdown listing all namespaces that the current user has access to.
+ *  The user can select a namespace from the dropdown to navigate directly to that namespace.
+ *  The "Manage" button directs the user to the namespace management page.
+ *  The "Refresh List" button refrehes the list of namespaces in the dropdown.
+ *
+ * @example
+ * <NamespacePicker class="hds-side-nav-hide-when-minimized" />
+ */
 
-  actions: {
-    refreshNamespaceList() {
-      this.namespaceService.findNamespacesForUser.perform();
-    },
-  },
-});
+export default class ApplicationComponent extends Component {
+  @service namespace;
+
+  @tracked selected = {};
+  @tracked options = [];
+
+  constructor() {
+    super(...arguments);
+    this.loadOptions();
+  }
+
+  #matchesPath(option, currentNamespace) {
+    // TODO: Revisit. A hardcoded check for "path" & "/path" seems hacky, but it fixes a breaking test:
+    //  "Acceptance | Enterprise | namespaces: it shows nested namespaces if you log in with a namespace starting with a /"
+    //  My assumption is that namespace shouldn't start with a "/", but is this a HVD thing? or is the test outdated?
+    return option?.path === currentNamespace?.path || `/${option?.path}` === currentNamespace?.path;
+  }
+
+  #getSelected(options, currentNamespace) {
+    return options.find((option) => this.#matchesPath(option, currentNamespace));
+  }
+
+  #getOptions(namespace) {
+    /* Each namespace option has 3 properties: { id, path, and label }
+     *   - id: node / namespace name (displayed when the namespace picker is closed)
+     *   - path: full namespace path (used to navigate to the namespace)
+     *   - label: text displayed inside the namespace picker dropdown (if root, then label = id, else label = path)
+     *
+     *  Example:
+     *   | id         | path                    | label                   |
+     *   | ---        | ----                    | -----                   |
+     *   | root       | ''                      | 'root'                  |
+     *   | namespace1 | 'namespace1'            | 'namespace1'            |
+     *   | namespace2 | 'namespace1/namespace2' | 'namespace1/namespace2' |
+     */
+    return [
+      // TODO: Some users (including HDS Admin User) should never see the root namespace. Address this in a followup PR.
+      { id: 'root', path: '', label: 'root' },
```

Comment:
Note: Added a TODO to revisit this,`root` should not be hardcoded. Will address this in a separate PR.